### PR TITLE
Update Nitro to C++17

### PR DESCRIPTION
**Pull request by abellgithub**

### CODE REVIEW

File: c/CMakeLists.txt [hunk 1]
```diff
@@ -13,10 +13,7 @@ add_sources(nitf/shared/*.c)
 
 add_library(${NITF_C_LIB_NAME} SHARED ${SRC})
```
Comment (hobu) [inline c/CMakeLists.txt]: should we do PRIVATE target_include_dir and target_link_libraries here too?

File: c/nitf/include/nitf/System.h [hunk 2]
```diff
@@ -62,15 +62,15 @@
...
 #ifdef NITF_DEBUG
     #define NRT_DEBUG 1
-    #include "nrt/Debug.h"
-
-    #define NITF_MEM_LOG        NRT_MEM_LOG
-    #define nitf_Debug_malloc   nrt_Debug_malloc
-    #define nitf_Debug_realloc  nrt_Debug_realloc
-    #define nitf_Debug_free     nrt_Debug_free
 #endif
+
```
Comment (hobu) [inline c/nitf/include/nitf/System.h]: Why were these moved out of debug?

**Review by hobu — APPROVED** [review]

File: c/nitf/include/nitf/System.h (hunk 2 above)
Comment (abellgithub) [reply c/nitf/include/nitf/System.h]: They need to be defined, even if they don't do anything.  In modern C, you have to declare a function.  In the old days, you didn't have to.

File: c/CMakeLists.txt (hunk 1 above)
Comment (abellgithub) [reply c/CMakeLists.txt]: I don't understand.  This is just creating the C library from source files.
